src/core/notation.js

```diff
@@ -412,7 +419,7 @@ class Notation {
      *  console.log(obj);
      *  // { notebook: "Mac", car: { brand: "Ford", model: "Mustang", year: 1970, color: "red" }, boat: "none" };
      */
-    set (notation, value, overwrite = true) {
+    set (notation, value, overwrite = true, concatArrays = false) {
```

Comment:
Why are you changing the signature?

Reply:
It was the quickest way to fix merging notations.
Here is code from tests you attached:
```
    it('Notation #merge()', () => {
        const nota = create({ a: [{ x: 1 }], b: { c: [3] } });
        let merged = nota.merge({ a: [{ y: 2 }] });
        expect(merged.value.a[0]).toEqual({ x: 1, y: 2 });
        merged = nota.merge({ b: { c: [4, 5] } });
        expect(merged.value.b.c).toEqual([3, 4, 5]);
    });
```
Without array concatenation last expect would expect value `[4, 5]`  to be equal to `[3, 4, 5]` because both `[3]` and `[4, 5]` had their `0` index taken. So second array elements would overwrite first array elements.